core/embed/trezorhal/stm32u5/layout.c

```diff
@@ -1,82 +1,38 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
 #include "flash.h"
+#include "layout_helpers.h"
 #include "model.h"
 
-const flash_area_t STORAGE_AREAS[STORAGE_AREAS_COUNT] = {
-    {
-        .num_subareas = 1,
-        .subarea[0] =
-            {
-                .first_sector = STORAGE_1_SECTOR_START,
-                .num_sectors =
-                    STORAGE_1_SECTOR_END - STORAGE_1_SECTOR_START + 1,
-            },
-    },
-    {
-        .num_subareas = 1,
-        .subarea[0] =
-            {
-                .first_sector = STORAGE_2_SECTOR_START,
-                .num_sectors =
-                    STORAGE_2_SECTOR_END - STORAGE_2_SECTOR_START + 1,
-            },
-    },
-};
-
-const flash_area_t BOARDLOADER_AREA = {
-    .num_subareas = 1,
-    .subarea[0] =
-        {
-            .first_sector = BOARDLOADER_SECTOR_START,
-            .num_sectors =
-                BOARDLOADER_SECTOR_END - BOARDLOADER_SECTOR_START + 1,
-        },
-};
-
-const flash_area_t BOOTLOADER_AREA = {
-    .num_subareas = 1,
-    .subarea[0] =
-        {
-            .first_sector = BOOTLOADER_SECTOR_START,
-            .num_sectors = BOOTLOADER_SECTOR_END - BOOTLOADER_SECTOR_START + 1,
-        },
-};
-
-const flash_area_t FIRMWARE_AREA = {
-    .num_subareas = 1,
-    .subarea[0] =
-        {
-            .first_sector = FIRMWARE_SECTOR_START,
-            .num_sectors = FIRMWARE_SECTOR_END - FIRMWARE_SECTOR_START + 1,
-        },
-};
-
-const flash_area_t SECRET_AREA = {
-    .num_subareas = 1,
-    .subarea[0] =
-        {
-            .first_sector = SECRET_SECTOR_START,
-            .num_sectors = SECRET_SECTOR_END - SECRET_SECTOR_START + 1,
-        },
-};
-
-const flash_area_t BHK_AREA = {
-    .num_subareas = 1,
-    .subarea[0] =
-        {
-            .first_sector = BHK_SECTOR_START,
-            .num_sectors = BHK_SECTOR_END - BHK_SECTOR_START + 1,
-        },
-};
-
-const flash_area_t ASSETS_AREA = {
-    .num_subareas = 1,
-    .subarea[0] =
-        {
-            .first_sector = ASSETS_SECTOR_START,
-            .num_sectors = ASSETS_SECTOR_END - ASSETS_SECTOR_START + 1,
-        },
-};
-
-const flash_area_t UNUSED_AREA = {
-    .num_subareas = 0,
-};
+// Convert sector number to address
+//
+// This conversion is used in static assert in definitions below
+#define FLASH_SECTOR_TO_ADDR(sector) (FLASH_BASE_S + (sector) * 8192)
```

Comment:
you can reference FLASH_PAGE_SIZE here

Reply:
yes, I changed it, see f31168508f7e2927b35368011c82bbe83065d97e